packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -312,71 +489,122 @@ export default {
-          eventBranches.some((branch) => e.id >= branch[0] && e.id <= branch[1])
-        );
+      if (this.filters.declutter.hideMediaRequests.on) {
+        events = this.filterMediaRequests(events);
+      }
 
-        // it's necessary to remove package wrap in Component Diagram,
-        // without this code package will be rendered in collapsed state
-        if (
-          this.rootCodeObject.type === 'class' &&
-          this.rootCodeObject.packageObject
-        ) {
-          classMap = new ClassMap([
-            classMap.codeObjects.filter(
-              (co) => co.id !== this.rootCodeObject.packageObject.id
-            ),
-          ]);
-        }
+      if (this.filters.declutter.hideUnlabeled.on) {
+        events = events.filter(
+          (e) =>
+            e.labels.size > 0 || e.codeObject.type !== CodeObjectType.FUNCTION
+        );
       }
 
-      events = events.filter(
-        (e) =>
-          this.filters.unlabeledEvents.on ||
-          e.labels.size > 0 ||
-          e.codeObject.type !== CodeObjectType.FUNCTION
-      );
+      if (
+        this.filters.declutter.hideElapsedTimeUnder.on &&
+        this.filters.declutter.hideElapsedTimeUnder.time > 0
+      ) {
+        events = events.filter(
+          (e) =>
+            e.returnEvent &&
+            e.returnEvent.elapsedTime &&
+            e.returnEvent.elapsedTime >=
+              this.filters.declutter.hideElapsedTimeUnder.time / 1000
+        );
+      }
 
-      if (this.filters.hideMediaRequests.on) {
-        const mediaRegex = [
-          'application/javascript',
-          'application/ecmascript',
-          'audio/.+',
-          'font/.+',
-          'image/.+',
-          'text/javascript',
-          'text/ecmascript',
-          'text/css',
-          'video/.+',
-        ].map((t) => new RegExp(t, 'i'));
-        const excludedEvents = [];
-        events.forEach((e) => {
-          if (
-            e.http_server_response &&
-            e.http_server_response.mime_type &&
-            mediaRegex.some((regex) =>
-              regex.test(e.http_server_response.mime_type)
-            )
-          ) {
-            excludedEvents.push(e.parent_id);
+      if (
+        this.filters.declutter.hideName.on &&
+        this.filters.declutter.hideName.names.length
+      ) {
+        this.filters.declutter.hideName.names.forEach((fqid) => {
+          const codeObject = classMap.codeObjects.filter(
+            (co) => co.fqid === fqid
+          )[0];
+          if (codeObject) {
+            events = events.filter((e) => !codeObject.allEvents.includes(e));
           }
         });
-        events = events.filter((e) => !excludedEvents.includes(e.id));
       }
 
+      const eventIds = events.filter((e) => e.isCall()).map((e) => e.id);
+
       return buildAppMap({
-        events,
+        events: events.filter(
+          (e) =>
+            (e.isCall() && eventIds.includes(e.id)) ||
+            eventIds.includes(e.parent_id)
```

Comment:
Worst case complexity using an array for `eventIds` here is O(2n^2).

`eventIds` should be a `Set`.